### DISCUSSION TITLE
Add Mega Watt (MW) power unit

### DISCUSSION THREAD

**Discussion by mdallaire** [home-assistant/architecture#917] [Entity Models] [answered]
# Proposal
Add unit MW (Mega Watt) to the power device class.

# Background
- Some electricity providers publish their whole grid power demand in MW
- Without support for the MW unit the sensor generate errors in Home-Assistant logs.

# Initial implementation
Core PR: https://github.com/home-assistant/core/pull/94358
Developer docs PR: https://github.com/home-assistant/developers.home-assistant/pull/1794

**Comment by frenck** [comment]
This means we are adding constants and support for a unit. Which integrations can currently use these?
In general, we do want at least a couple of integrations using this, before we can/want to make it part of core.

../Frenck

**Comment by frenck** [comment, marked as answer]
I've discussed this request with a couple of core members, and at this time, we don't see a bigger use-case to add this, nor are there any integrations that can currently provide such values.

For that reason, we are not going to add it at this point in time.

../Frenck

**Reply by amordicus** [reply]
I use unit MW similarly to [mdallaire] and [joriws](https://github.com/joriws) but, I understand that it is a question of priority and it can be low in the list of issues.

Probably people make use of MW and just ignore the warning in the log file.  

I like the solution suggested in issue #99061 raised against homeassistant/core. More work today but, less on the long run.

Text copied from issue #99061 "Maybe home assistant should be handling SI prefixes like n(ano),u(micro),m(illi), none, k(ilo), M, G, T etc, and separate unit W - watts and understand prefixes just as 10^X."

Perhaps there is a work around the warnings that could be suggested here? What if we don't use the "device_class: power" would it work?
